cmd/release-job-migrator/main.go

```diff
@@ -0,0 +1,460 @@
+			return filepath.SkipDir
+		}
+		if !strings.HasSuffix(path, "-periodics.yaml") {
+			return nil
+		}
+		raw, err := ioutil.ReadFile(path)
+		if err != nil {
+			return fmt.Errorf("failed to read file %s: %w", info.Name(), err)
+		}
+		periodics := prowconfig.JobConfig{}
+		if err := yaml.UnmarshalStrict(raw, &periodics); err != nil {
+			return fmt.Errorf("failed to unmarshal file %s: %w", info.Name(), err)
+		}
+		jobs[info.Name()] = periodics
+		return nil
+	}); err != nil {
+		return fmt.Errorf("failed to load periodic job configs: %w", err)
+	}
+
+	rawConfig, err := ioutil.ReadFile(o.config)
+	if err != nil {
+		return fmt.Errorf("failed to read config file %s: %w", o.config, err)
+	}
+	generatorConfig := GeneratorConfig{}
+	if err := yaml.UnmarshalStrict(rawConfig, &generatorConfig); err != nil {
+		return fmt.Errorf("failed to unmarshal config file %s: %w", o.config, err)
+	}
+	ciopConfigs, err := configlib.LoadDataByFilename(o.ciopConfigDir)
+	if err != nil {
+		return fmt.Errorf("failed to load ci-operator configs: %w", err)
+	}
+
+	// store replacement info for each ci-op config
+	replacements := make(map[string]testsAndBaseImages)
+	// key: old jobname, value: new (generated jobname)
+	replacedJobs := make(map[string]string)
+	// list of test names for detected release jobs with no configuration
+	configlessTests := sets.NewString()
+	for _, jobConfig := range jobs {
+		for _, periodic := range jobConfig.Periodics {
+			info, isReleaseJob := getJobInfo(periodic.Name)
+			if !isReleaseJob {
+				continue
+			}
+			// avoid jobs that do more than just run ci-operator
+			if periodic.Spec.Containers[0].Command[0] != "ci-operator" {
+				logrus.Warnf("periodic job %s has a command != \"ci-operator\", ignoring", periodic.Name)
+				continue
+			}
+			// check if an UNRESOLVED_CONFIG exists and populate an env var map to handle configs with env vars
+			envs := make(map[string]string)
+			isUnresolved := false
+			for _, env := range periodic.Spec.Containers[0].Env {
+				envs[env.Name] = env.Value
+				if env.Name == "UNRESOLVED_CONFIG" {
+					isUnresolved = true
+				}
+			}
+			filename := metadataFromJobInfo(info).Basename()
+			if _, ok := replacements[filename]; !ok {
+				replacements[filename] = testsAndBaseImages{
+					baseImages: make(map[string]api.ImageStreamTagReference),
+				}
+			}
+			var config testConfig
+			if isUnresolved {
+				target := ""
+				for _, arg := range periodic.Spec.Containers[0].Args {
+					if strings.Split(arg, "=")[0] == "--target" {
+						target = strings.Split(arg, "=")[1]
+						// args can contain env vars; make sure they're replaced
+						for name, value := range envs {
+							target = strings.ReplaceAll(target, fmt.Sprintf("$(%s)", name), value)
+						}
+					}
+				}
+				if target == "" {
+					return fmt.Errorf("found UNRESOLVED_CONFIG for job %s but could not identify target job", periodic.Name)
+				}
+				// replace all env vars in UNRESOLVED CONFIG
```

Comment:
note: the code handles this but IIRC there are jobs that dynamically edit their own config before invoking `ci-operator` for which the intent may have been to dynamically generate the config? in any case we will handle it later ...

Reply:
This shouldn't be a problem for the jobs that this tool will handle. It's configured to only migrate jobs that use `ci-operator` as the command and simply print a warning notifying the user that that job will be ignored if not. (https://github.com/openshift/ci-tools/pull/1450/files#diff-fff22e1ebd0d1f2036370917bb8fc4706915281da9fcd2c6f2b15cdd36eabca2R264-R267)